docs/DeveloperGuide.md

```diff
@@ -184,6 +184,66 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+
+##### Updating Filtered Person List:
+
+The `ModelManager` class implements the `Model` interface and manages the application's data. It maintains a `filteredPersons` list, which is a FilteredList of contacts based on the applied predicate. The `updateFilteredPersonList` method implemented in `ModelManager` updates the filtered list based on the provided predicate.
+
+When the `FindCommand` is executed, the `updateFilteredPersonList` method is called with the `PersonDetailContainsKeywordPredicate` as a parameter. This updates the `filteredPersons` list to show only persons that fufill the predicate.
+
+##### User Interface Interaction
+
+After the `filteredPersons` list is updated, the user interface is updated such that the `PersonListPanel` now shows persons the fufill the predicate generated by the original user input.
+
+The following sequence diagram illustrates the `find` command with the user input `find n/Alice`
+<puml src="diagrams/FindImplementationSequenceDiagram.puml" width="550" />
+
+The following activity Diagram illustrates the user execution of the `find` command
+<puml src="diagrams/FindImplementationActivityDiagram.puml" width="550" />
+
+#### **Considerations**
+
+##### User Interface Consistency
+
+The choice of implementing the command to use prefixes to determine the filter criteria ensures consistency with other commands in the application. As this command follows a similar structure to all other commands, it is easier for users to learn and use the application.
+
+##### Flexibility in Search Criteria
+
+By allowing users to specify search criteria using different prefixes (name, phone, email, etc.), the implementation offers flexibility.
+Users can search for contacts based on various details, enhancing the usability of the feature. In the context of our potential users, we considered that users would likely have to sometimes filter students by their classes, or filter people by their roles (student, tutor, professor). So we opted to implement this feature with the flexibility of using all prefixes to account for all these potential use cases.
+
+##### Predicate-based Filtering
+
+As the `Model` class was built prior to the implementation of this feature, we did our best to re-use available methods instead of unnecessarily re-programing already exisiting logic. Hence, we decided to craft the command around the idea of a custom predicate as the `Model` class already had a `updateFilteredPersonList` method implemented that would filter persons using a predicate.
+
+##### Extensibility
+
+This design allows for easy extension to accommodate future enhancements or additional search criteria. New prefixes can be added to support additional search criteria without significant changes as we merely need to update our `Predicate` logic. This ensures that the implementation remains adaptable to evolving requirements and we can upgrade and improve the feature whenever required.
+
```

Comment:
Very detailed explanation of the find command! It may even help readers find themselves.

Reply:
Yes you have really considered a lot of factors this would allow us to define our expected behaviors of the find command, reducing bugs too. Good Job!